src/devices/bresser_lightning.c

```diff
@@ -0,0 +1,133 @@
+/** @file
+    Bresser Lightning Sensor.
+
+    Copyright (C) 2023 The rtl_433 Project
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+*/
+
+#include "decoder.h"
+
+#define SENSOR_TYPE_LIGHTNING 9
+
+/**
+Bresser Lightning Sensor.
+
+Decoder for Bresser lightning outdoor sensor, PN 7009976
+
+see https://github.com/merbanan/rtl_433/issues/2140
+
+Preamble: aa aa 2d d4
+
+Data layout:
+    DIGEST:8h8h ID:8h8h CTR:12h BATT:1b ?3b STYPE:4h STARTUP:1b CH:3d KM:8d ?8h8h
+
+Based on bresser_7in1.c
+
+The data (not including STYPE, STARTUP, CH and maybe ID) has a whitening of 0xaa.
+CH is always 0.
+
+First two bytes are an LFSR-16 digest, generator 0x8810 key 0xabf9 with a final xor 0x899e
+*/
+
+static int bresser_lightning_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    uint8_t const preamble_pattern[] = {0xaa, 0xaa, 0x2d, 0xd4};
+    uint8_t msg[25];
+
+    if (bitbuffer->num_rows != 1
+            || bitbuffer->bits_per_row[0] < 160
+            || bitbuffer->bits_per_row[0] > 440) {
+        decoder_logf(decoder, 2, __func__, "bit_per_row %u out of range", bitbuffer->bits_per_row[0]);
+        return DECODE_ABORT_EARLY; // Unrecognized data
+    }
+
+    unsigned start_pos = bitbuffer_search(bitbuffer, 0, 0,
+            preamble_pattern, sizeof (preamble_pattern) * 8);
+
+    if (start_pos >= bitbuffer->bits_per_row[0]) {
+        return DECODE_ABORT_LENGTH;
+    }
+    start_pos += sizeof (preamble_pattern) * 8;
+
+    unsigned len = bitbuffer->bits_per_row[0] - start_pos;
+    if (len < sizeof(msg) * 8) {
+        decoder_logf(decoder, 2, __func__, "%u too short", len);
+        return DECODE_ABORT_LENGTH; // message too short
+    }
+
+    bitbuffer_extract_bytes(bitbuffer, 0, start_pos, msg, sizeof(msg) * 8);
+
+    decoder_log_bitrow(decoder, 2, __func__, msg, sizeof(msg) * 8, "MSG");
+
+    int s_type      = msg[6] >> 4;
+    int chan        = msg[6] & 0x07;
+    int battery_ok  = (msg[5] & 0x08) ? 0 : 1;
```

Comment:
we'd usually use a shift instead of a ternary, i.e. `>> 3` and then name it `battery_low`

Reply:
changed, also for `startup` (renamed `nstartup`)